tests/integration/security/sds_tls_origination/egress_gateway_origination_test.go

```diff
@@ -61,66 +61,69 @@ func TestSimpleTlsOrigination(t *testing.T) {
 			// Set up Host Namespace
 			host := sdstlsutil.ServerSvc + "." + apps.ServerNamespace.Name() + ".svc.cluster.local"
 
-			testCases := map[string]sdstlsutil.TestCase{
+			testCases := []sdstlsutil.TestCase{
 				// Use CA certificate stored as k8s secret with the same issuing CA as server's CA.
 				// This root certificate can validate the server cert presented by the echoboot server instance.
-				"Simple TLS with Correct Root Cert": {
+				{
+					Name:            "Simple TLS with Correct Root Cert",
 					Response:        response.StatusCodeOK,
 					CredentialToUse: strings.TrimSuffix(credName, "-cacert"),
 					Gateway:         true,
 				},
 				// Use CA certificate stored as k8s secret with different issuing CA as server's CA.
 				// This root certificate cannot validate the server cert presented by the echoboot server instance.
-				"Simple TLS with Fake Root Cert": {
+				{
+					Name:            "Simple TLS with Fake Root Cert",
 					Response:        response.StatusCodeUnavailable,
 					CredentialToUse: strings.TrimSuffix(fakeCredName, "-cacert"),
 					Gateway:         false,
 				},
 
 				// Set up an UpstreamCluster with a CredentialName when secret doesn't even exist in istio-system ns.
 				// Secret fetching error at Gateway, results in a 503 response.
-				"Simple TLS with credentialName set when the underlying secret doesn't exist": {
+				{
+					Name:            "Simple TLS with credentialName set when the underlying secret doesn't exist",
 					Response:        response.StatusCodeUnavailable,
 					CredentialToUse: strings.TrimSuffix(credNameMissing, "-cacert"),
 					Gateway:         false,
 				},
 			}
 
-			for name, tc := range testCases {
-				echotest.New(t, apps.All).
-					SetupForDestination(func(t framework.TestContext, dst echo.Instances) error {
-						bufDestinationRule := sdstlsutil.CreateDestinationRule(t, apps.ServerNamespace, "SIMPLE", tc.CredentialToUse)
+			for _, tc := range testCases {
+				t.NewSubTest(tc.Name).Run(func(t framework.TestContext) {
+					echotest.New(t, apps.All).
+						SetupForDestination(func(t framework.TestContext, dst echo.Instances) error {
```

Comment:
Seems like we don't use `tc` or `dst` in `SetupForDestination`. Can we move it outside the loop?